mtags/src/main/scala-2/scala/meta/internal/pc/HoverProvider.scala

```diff
@@ -13,89 +15,115 @@ import org.eclipse.lsp4j.MarkupKind
+        )
+        val tree = typedHoverTreeAt(pos)
+        (pos, tree)
+      case params: OffsetParams =>
+        val pos = unit.position(params.offset())
+        val tree = typedHoverTreeAt(pos)
+        (pos, tree)
+    }
+    tree match {
+      case i @ Import(_, _) =>
+        for {
+          member <- i.selector(pos)
+          hover <- toHover(member, pos)
+        } yield hover
+      case _: Select | _: Apply | _: TypeApply | _: Ident =>
+        val expanded = expandRangeToEnclosingApply(pos)
+        if (
+          expanded != null &&
+          expanded.tpe != null &&
+          tree.symbol != null &&
+          expanded.symbol != null
+        ) {
+          val symbol =
+            if (expanded.symbol.isConstructor) expanded.symbol
+            else tree.symbol
           toHover(
             symbol,
-            v.symbol.keyString,
-            symbol.info,
-            symbol.info,
+            symbol.keyString,
+            seenFromType(tree, symbol),
+            expanded.tpe,
             pos,
-            v.pos
+            expanded.pos
           )
-        // Bound variable in a pattern match, example `head` in `case head :: tail =>`
-        case _: Bind =>
-          val symbol = tree.symbol
-          toHover(
-            symbol = symbol,
-            keyword = "",
-            seenFrom = symbol.info,
-            tpe = symbol.info,
-            pos = pos,
-            range = pos
-          )
-        case _ =>
-          // Don't show hover for non-identifiers.
-          None
-      }
+        } else {
+          for {
+            sym <- Option(tree.symbol)
+            tpe <- Option(tree.tpe)
+            seenFrom = seenFromType(tree, sym)
+            hover <- toHover(sym, sym.keyString, seenFrom, tpe, pos, tree.pos)
+          } yield hover
+        }
+      case UnApply(fun, _) if fun.symbol != null =>
+        toHover(
+          fun.symbol,
+          fun.symbol.keyString,
+          seenFromType(tree, fun.symbol),
+          tree.tpe,
+          pos,
+          pos
+        )
+      // Def, val or val definition, example `val x: Int = 1`
+      // Matches only if the cursor is over the definition name.
+      case v: ValOrDefDef
+          if (v.namePos
+            .includes(pos) || pos.includes(v.namePos)) && v.symbol != null =>
+        val symbol = (v.symbol.getter: Symbol) match {
+          case NoSymbol => v.symbol
+          case getter => getter
+        }
+        toHover(
+          symbol,
+          v.symbol.keyString,
+          symbol.info,
+          symbol.info,
+          pos,
+          v.pos
+        )
+      // Bound variable in a pattern match, example `head` in `case head :: tail =>`
+      case _: Bind =>
+        val symbol = tree.symbol
+        toHover(
+          symbol = symbol,
+          keyword = "",
+          seenFrom = symbol.info,
+          tpe = symbol.info,
+          pos = pos,
+          range = pos
+        )
+      case _: Literal =>
```

Comment:
```suggestion
      case _: Literal if params.isInstanceOf[RangeParams] =>
```
or something along these lines in order not to change the current behaviour.

Reply:
ok